Validation generation and `private static final` fields

Hola!

If you have a `module-info.java` or `package-info.java` with an `@NullMarked` JSpecify annotation, and then have a record that looks like this:

```java
@Valid
public record Insect(
  @NotBlank @Size(min = 1, max = 50) String name
) {
  private static final Set<String> FLYING = Set.of("Fly", "Butterfly");
  private static final Set<String> WALKING = Set.of("Ant", "Stick insect");

  public Set<String> associated() {
    if (FLYING.contains(name)) {
      return FLYING;
    }
    return WALKING;
  }
}
```

The generator errors out with the following:

```
[ERROR] Non accessible field example.Insect FLYING with no matching getter?
[ERROR] Non accessible field example.Insect WALKING with no matching getter?
[ERROR] Error ValidationAdapter due to nonAccessibleField for example.Insect
```

Could the generator please ignore `private static final` fields?

--- a/validator-generator/src/main/java/io/avaje/validation/generator/TypeReader.java
+++ b/validator-generator/src/main/java/io/avaje/validation/generator/TypeReader.java
@@ -108,7 +108,9 @@ final class TypeReader {
       element = mixInField;
     }
 
-    if (includeField(element) || Util.isNonNullable(element)) {
+    if (includeField(element)
+        || !element.getModifiers().contains(Modifier.STATIC)
+            && Util.isNonNullable(element)) {
       seenFields.add(element.toString());
       var reader = new FieldReader(element, genericTypeParams);
       if (reader.hasConstraints() || ValidPrism.isPresent(element)) {

--- a/validator-generator/src/test/java/io/avaje/validation/generator/models/valid/Insect.java
+++ b/validator-generator/src/test/java/io/avaje/validation/generator/models/valid/Insect.java
@@ -1,0 +1,23 @@
+package io.avaje.validation.generator.models.valid;
+
+import java.util.Set;
+
+import org.jspecify.annotations.NullMarked;
+
+import io.avaje.validation.constraints.NotBlank;
+import io.avaje.validation.constraints.Size;
+import jakarta.validation.Valid;
+
+@Valid
+@NullMarked
+public record Insect(@NotBlank @Size(min = 1, max = 50) String name) {
+  private static final Set<String> FLYING = Set.of("Fly", "Butterfly");
+  private static final Set<String> WALKING = Set.of("Ant", "Stick insect");
+
+  public Set<String> associated() {
+    if (FLYING.contains(name)) {
+      return FLYING;
+    }
+    return WALKING;
+  }
+}

